modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -153,6 +153,18 @@ object CrossSources {
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
 
+    val preprocessedWithUsingDirs =
+      preprocessedInputFromArgs.filter(p => p.options.exists(_ != BuildOptions()))
+    if (preprocessedWithUsingDirs.length > 1)
+      inputs.elements.projectSettingsFiles.headOption match {
+        case Some(p) => logger.message(
+            s"Warning: using directives detected in multiple files. It is recommended to keep them centralized in the ${p.path} file."
```

Comment:
Shouldn't we report this problem as warnings where the directives are  defined excluding project file itself?

Reply:
Good point, didn't think about it. I will change that.

Reply:
We should always raise warning, if there is more than one file with using directives (no matter if some of them are project files). 

As far as I know, we shouldn't raise warning if all using directives are at most in one file (even if it is not project file).

Reply:
Are you sure we should exclude project file from the warning?